snapshot/snapshotfs/upload.go

```diff
@@ -550,7 +555,7 @@ func (u *Uploader) processUploadWorkItems(workItems []*uploadWorkItem, dirManife
 		}
 
 		if result.err != nil {
-			if u.IgnoreFileErrors {
+			if ignoreFileErrs {
```

Comment:
All types of both directory and file errors are being ignored when the appropriate values are set in the policy.
I'm wondering whether or not it makes sense to differentiate permission errors from other types of errors when ignoring them. This is not suggesting adding more flags to the policy, but instead have those flags only control permission errors and alway fail in other cases, such as not being able to read (parts of) a file or a directory due to an I/O error such as media corruption.
In the interest of addressing the upstream issue, perhaps we should push this and iterate in a separate PR.

Reply:
Took a quick stab at this. Pretty trivial to add `errors.Is(err, os.ErrPermission)` but I think I'll punt on it for now: adding it blindly will slightly alter the behavior of migration commands from what it is today. Maybe we want that, but I think it needs more consideration.

Reply:
Yes, it makes sense to see what the ramifications are and whether or not it is needed and worth it.